### ISSUE TITLE
[BUG] Hiding the BC Launcher crashes BC

### ISSUE THREAD

**Issue by lgmctimmer** [bitfocus/companion#2692]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I can run Bitfocus Companion just fine. However, while I'm using the app, I want to hide the background launcher. So I click the conveniently labelled "hide" button. Unfortunately, if I want to quit the BC browser window and then reopen it later, I cannot. I try looking for the BC launcher that I previously hid, but I can't find it anywhere. It's not in my dock, it's not alt-tabbable, and it won't even reopen when I click on the desktop application. I can find it running in my background if I pull up my activity monitor. I'm able to quit the background instance of BC and reopen it on the desktop. However, when I reopen BC, it is no longer able to find a server. It is perpetually stuck on "unknown waiting for webserver" and I'm forced to restart my computer for it to be able to work normally again.

### Steps To Reproduce

1. Open Companion App on the Desktop
2. Hide Companion App
3. Open Activity Monitor
4. Force Quit Companion App running in the Background
5. Open Companion App on Desktop

### Expected Behavior

When I hide the Companion App, I expect to be able to reopen it again somehow. I don't know where to find the hidden window.

### Environment (please complete the following information)

```markdown
- OS: Mac Ventura 13.2.1
- Browser: Chrome
- Companion Version: 3.1.2
```

**Comment by Julusian** [comment]
When this happens, what is the process in activity monitor called?
Before you press hide, does it show up in the tray? If so, does it disappear as soon as you hide it, or does it disappear later on?

I'm not able to reproduce this on my mac (running macos 14), so don't know where to begin on this. Perhaps this is related to #2720?

**Comment by Julusian** [comment]
@lgmctimmer any chance of some answers to my questions? without some help I don't think anything can be done on this

**Comment by Julusian** [comment]
If this wasn't fixed by #2720, let me know and this can be reopened
